ui/components/accounts/AccountsListRow.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+  <oc-table-row>
+    <oc-table-cell>
+      <avatar :user-name="account.displayName || account.onPremisesSamAccountName" :userid="account.id" :width="35" />
+    </oc-table-cell>
+    <oc-table-cell v-text="account.onPremisesSamAccountName" />
+    <oc-table-cell v-text="account.displayName || '-'" />
+    <oc-table-cell v-text="account.mail" />
+    <oc-table-cell>
+      <oc-button :class="`accounts-roles-select-trigger-${account.id}`" variation="raw">
+        <span class="uk-flex uk-flex-middle">
+          {{ currentRole ? currentRole.displayName : $gettext('Select role') }}
+          <oc-icon name="expand_more" aria-hidden="true" />
+        </span>
+      </oc-button>
+      <oc-drop
+        :drop-id="`accounts-roles-select-dropdown-${account.id}`"
+        :toggle="`.accounts-roles-select-trigger-${account.id}`"
+        mode="click"
+        close-on-click
+        :options="{ delayHide: 0 }"
+      >
+        <ul class="uk-nav">
+          <li v-for="role in roles" :key="role.id" class="uk-margin-small">
+            <oc-button variation="raw" @click="changeRole(role.id)">
+              <span v-text="role.displayName" :class="{ 'uk-text-bold': role === currentRole }" />
```

Comment:
the only thing I can find for improvement is, that the drop could use a visual indicator for the already selected role - not just the bold text, but e.g. a clicked radiobutton. My original idea a while ago was, to bring all the form elements from ocis-settings into the ODS and reuse them all over the place. In ocis-settings there is a single choice dropdown with radiobuttons and a multi choice dropdown with checkboxes. Maybe it's possible to reuse the single choice dropdown with radiobuttons here, and then later on bring a more generic implementation into ODS and use it whereever possible.
See [ocis-settings single-choice](https://github.com/owncloud/ocis-settings/blob/master/ui/components/settings/SettingSingleChoice.vue) for reference.

Reply:
@LukasHirt ^